src/java.base/share/native/libjli/java.c

```diff
@@ -454,6 +455,37 @@ JLI_Launch(int argc, char ** argv,              /* main argc, argv */
         } \
     } while (JNI_FALSE)
 
+static char *getCheckpointDir(int argc, char **argv) {
+    char *checkpointDir = NULL;
+    const char *target = "-XX:CRaCRestoreFrom=";
+    for (int i = 0; i < argc; i++) {
+        if (0 == strncmp(argv[i], target, strlen(target))) {
+            checkpointDir = (char *)JLI_MemAlloc(strlen(argv[i] + strlen(target)) + 1);
+            if (NULL != checkpointDir) {
+                strcpy(checkpointDir, argv[i] + strlen(target));
+            } else {
+                JLI_ReportErrorMessage("Failed to allocate memory for checkpointDir.");
+                exit(EXIT_FAILURE);
+            }
+        }
+    }
+    return checkpointDir;
+}
+
+static int restoreFromCheckpoint(char *checkpointDir) {
+    char* argv[7];
+    argv[0] = "criu";
+    argv[1] = "restore";
+    argv[2] = "-D";
+    argv[3] = checkpointDir;
+    argv[4] = "-v4";
+    argv[5] = "--shell-job";
+    argv[6] = NULL;
```

Comment:
there is a `--log-file` option as well that you can use `-Dopenj9.internal.criu.logFile` to detect.